### DISCUSSION TITLE
How to support concurrent in Genie?

### DISCUSSION THREAD

**Discussion by deahhh** [GenieFramework/Genie.jl#521] [Q&A] [answered]
$julia -q -t auto
```Julia
using Genie
route("/") do
   res = rand(1,10000000)*rand(10000000, 1)
   "$(Threads.threadid())\t$(res)"
end
up()
```
I tested with this code, but got 1 as the thread-id every time.
Does it support multi-threading?
Does it support multi-process?
Why it never stop even after I close the connection?("[ Info: GET / 200" will be printed several seconds after the http page was closed.)

**Comment by essenciary** [comment, marked as answer]
Genie execution is multi-tasked (non-blocking) but not multi-threaded (per the names in here https://docs.julialang.org/en/v1/manual/parallel-computing/). We're working on it though and if everything goes well, will be released in Genie 5 in a couple of weeks.

**Reply by deahhh** [reply]
@essenciary 
Does Genie 5 support multi-threading?
I find nothing in the document.
Thanks!
